src/loaders/ncbi_downloader/ncbi_downloader.py

```diff
@@ -1,15 +1,91 @@
+                          {202,207,80,83,86,89,95}
+                          [--root_dir ROOT_DIR]
+                          [--source SOURCE]
+                          [--threads THREADS] [--overwrite]
+
+options:
+  -h, --help            show this help message and exit
+
+required named arguments:
+  --download_file_ext DOWNLOAD_FILE_EXT [DOWNLOAD_FILE_EXT ...]
+                        Download only files that match given
+                        extensions.
+  --release_ver {202,207,80,83,86,89,95}
+                        GTDB release version
+
+optional arguments:
+  --root_dir ROOT_DIR   Root directory. (default: /global/cfs
+                        /cdirs/kbase/collections/sourcedata)
+  --source SOURCE       Source of data (default: GTDB)
+  --threads THREADS     Number of threads. (default: half of
+                        system cpu count)
+  --overwrite           Overwrite existing files.
+
+
+
+e.g.
+python ncbi_downloader.py --download_file_ext genomic.gff.gz genomic.fna.gz --release_ver 207
 
+NOTE:
+NERSC file structure for GTDB:
+/global/cfs/cdirs/kbase/collections/sourcedata/ -> GTDB -> [GTDB_release_version] -> [genome_id] -> genome files
+
+e.g.
+/global/cfs/cdirs/kbase/collections/sourcedata/GTDB -> r207 -> GB_GCA_000016605.1 -> genome files
+                                                            -> GB_GCA_000200715.1 -> genome files
+                                                       r202 -> GB_GCA_000016605.1 -> genome files
+                                                            -> GB_GCA_000200715.1 -> genome files
+                                                       r80  -> GB_GCA_000016605.1 -> genome files
+                                                            -> GB_GCA_000200715.1 -> genome files
+
+
+"""
+import argparse
+import itertools
+import multiprocessing
 import os
 import time
 from datetime import datetime
+from urllib import request
+from urllib.parse import urlparse
 
 import ftputil
+import requests
+from bs4 import BeautifulSoup
+
+# Fraction amount of system cores can be utilized
+# (i.e. 0.5 - program will use 50% of total processors,
+#       0.1 - program will use 10% of total processors)
+SYSTEM_UTILIZATION = 0.5
+SOURCE = ['GTDB']  # supported source of data
+GTDB_DOMAIN = 'https://data.gtdb.ecogenomic.org/releases/'
+
+
+def _parse_gtdb_release_vers():
+    # parse GTDB release versions from GTDB website (FTP port is closed)
+
+    response = requests.get(GTDB_DOMAIN)
+    soup = BeautifulSoup(response.text, 'html.parser')
 
+    links = soup.find_all('a')
 
-def _download_genome_file(download_dir: str, gene_id: str, target_file_ext: list[str]) -> None:
+    # release links follow format like 'latest/', 'release202/', 'release83/', etc.
+    release_ver = [link.get('href').split('release')[-1][:-1] for link in links if 'release' in link.get('href')]
+
+    return release_ver
+
+
+GTDB_RELEASE_VER = _parse_gtdb_release_vers()  # available GTDB release versions
```

Comment:
This will trigger an internet call every time this file is imported, pretty please don't do that. Move the call to `main()` at least and pass around the list as necessary

Reply:
👍